### PR TITLE
Use the short array syntax for the generated config.php during installation

### PR DESCRIPTION
**Pull request by jochenmanz**
### 1. Why is this change necessary?
The old php array syntax is used for the generated config.php during the installation process. This PR changes this behaviour

### 2. What does this change do, exactly?
The ConfigWriter uses the new and short array syntax for the config instead.

### 3. Describe each step to reproduce the issue or behaviour.
- install shopware without this change
- save the config.php
- install shopware with this change
- compare config.php

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by dneustadt** [comment]
Does this really justify another dependency, especially when needed exactly once?

**Comment by mitelg** [comment]
@dneustadt this is the `composer.json` of the installer. so maybe it is ok, because there is not that much code at all :wink:

**Comment by dneustadt** [comment]
@mitelg well according to githubs api the zend-code repo weighs in at 761348 bytes of PHP (https://api.github.com/repos/zendframework/zend-code/languages)
That's a whole lot of code in service to what arguably is code quality at best, even more in a single configuration file. I'd rather see a different file format for the config settings than carry a rather large package for the little benefit it provides.

**Comment by mitelg** [comment]
@dneustadt oh, I didn't know that :smile:  :see_no_evil: 
then you are totally right

**Comment by bcremer** [comment]
Please keep in mind that `zend-code` gets already pulled as an transitive dependency anyway:

```
~/www/shopware (5.4) 
$ composer install --no-dev --ignore-platform-reqs
Loading composer repositories with package information
Installing dependencies from lock file
Nothing to install or update
Generating optimized autoload files
Warning: Ambiguous class resolution, "Doctrine\ORM\Persisters\Entity\BasicEntityPersister" was found in both "$baseDir . '/engine/Library/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php" and "/home/bcremer/www/shopware/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php", the first will be used.
Warning: Ambiguous class resolution, "Doctrine\Common\Proxy\AbstractProxyFactory" was found in both "$baseDir . '/engine/Library/Doctrine/Common/Proxy/AbstractProxyFactory.php" and "/home/bcremer/www/shopware/vendor/doctrine/common/lib/Doctrine/Common/Proxy/AbstractProxyFactory.php", the first will be used.
~/www/shopware (5.4) 
$ composer show | grep zend-code                  
zendframework/zend-code         2.6.3     provides facilities to generate arbitrary code using an object oriented interface
```

Anyhow I'm also not seeing the benefit of using zend-code just for getting rid of old style arrays.

**Comment by jochenmanz** [comment]
I've reworked the whole thing now as you have a valid point. I added the socket to the build vars and replaced the zend-code approach with a str_replace approach leveraging the .dist file.

File: recovery/install/src/Service/ConfigWriter.php [hunk 1]
```diff
@@ -49,39 +50,31 @@ public function __construct($configPath)
...
         ];
 
-        foreach ($info as $key => $parameter) {
-            if ($key == 'port' && empty($parameter)) {
-                continue;
-            }
-            if ($key == 'socket' && empty($parameter)) {
-                continue;
-            }
+        $template = file_get_contents($this->configPath . '.dist');
```
Comment (shyim) [inline recovery/install/src/Service/ConfigWriter.php:66]: The release.zip does not contain a config.php.dist, this would fail or? @shopwareBot

**Pull request closed without merging**